[WIP] DRA e2e test to validate Nvidia gpu

tkashem opened the pull request.

tkashem commented:
/retest

tkashem committed:
add an e2e test to validate core DRA using the example driver

tkashem committed:
add an e2e test that validates the common spec

tkashem committed:
add an e2e that shows two distinct gpus are used

tkashem committed:
add an e2e, two containers sharing a gpu with TimeSlicing

tkashem committed:
add e2e that validates two containers sharing a gpu with MPS

tkashem committed:
add e2e, two gpus, one shared with MPS, the other with TimeSlicing

tkashem committed:
add e2e to validate pods can use MIG devices

tkashem committed:
add e2e to validate IPC with CUDA

tkashem committed:
add a dedicated test suite for dra

kannon92 commented on e2e-monitor-tests__20250915-155137.xml:
```diff
@@ -0,0 +1,10 @@
+<testsuite name="openshift-tests" tests="7" skipped="0" failures="0" time="0">
```
This looks like it should not be committed.

tkashem replied:
yes, removed

kannon92 commented on test-failures-summary_monitor_20250915-155137.json:
```diff
@@ -0,0 +1,23 @@
+{
```
Is this needed?

tkashem replied:
mistakenly added, i fixed it.

kannon92 commented on test/extended/dra/common_spec.go:
```diff
@@ -0,0 +1,115 @@
+	helper "github.com/openshift/origin/test/extended/dra/helper"
+
+	corev1 "k8s.io/api/core/v1"
+	resourceapi "k8s.io/api/resource/v1beta1"
```
/hold cancel

tkashem committed:
update go.mod

- use helm.sh/helm/v3 v3.18.0
- pin to the current version of
github.com/distribution/distribution/v3 being used

tkashem committed:
run go mod vendor

asahay19 committed:
Switch DRA beta API to v1

tkashem committed:
Merge pull request #2 from asahay19/asahay-pr-29842

Switch DRA beta API to v1
